blackbox/docs/sql/ddl.txt

```diff
@@ -1049,3 +1049,244 @@ means that if you re-create the table using the resulting ``CREATE TABLE``
 statement the settings of the 'old' table may differ from the settings of
 the 'new' table. This is because the table settings are set explicitly on
 creation time.
+
+User defined Functions
+======================
+
+CREATE OR REPLACE
+-----------------
+
+CrateDB supports user-defined functions. See :ref:`ref-create-function`
+for a full syntax description.
+
+These functions can be created, like so::
+
+     cr> CREATE FUNCTION add(integer, integer)
+     ...  RETURNS integer
+     ...  LANGUAGE JAVASCRIPT
+     ...  AS 'function(a, b){
+     ...        return a + b;
+     ...      }';
+
+``OR REPLACE`` can be used to replace an existing function::
+
+     cr> CREATE OR REPLACE FUNCTION add(integer, integer)
+     ...  RETURNS integer
+     ...  LANGUAGE JAVASCRIPT
+     ...  AS 'function(a, b){
+     ...        return a + b;
+     ...      }';
+
+Arguments can be named in the function definition.
+
+For example, if you wanted to name two ``geo_point`` arguments ``start`` and
+``end`` you would do it like so::
+
+     cr> CREATE OR REPLACE FUNCTION distance(start geo_point, end geo_point)
```

Comment:
this example will break as `distance` is already defined: https://crate.io/docs/reference/sql/scalar.html#distance-geo-point1-geo-point2-returns-double